cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -0,0 +1,1128 @@
+  {
+    static_assert(::cuda::std::is_same<T, It>::value || ::cuda::std::is_same<T, PS>::value, "");
+    _CCCL_DIAG_PUSH
+    _CCCL_DIAG_SUPPRESS_GCC("-Wstrict-aliasing")
+    return *reinterpret_cast<T*>(storage);
+    _CCCL_DIAG_POP
+  }
+};
+
+template <typename It>
+_CCCL_HOST_DEVICE auto make_iterator_kernel_arg(It it) -> kernel_arg<It>
+{
+  kernel_arg<It> arg;
+  cub::detail::uninitialized_copy_single(&arg.template aliased_storage<It>(), it);
+  return arg;
+}
+
+template <typename It>
+_CCCL_HOST_DEVICE auto make_aligned_base_ptr_kernel_arg(It ptr, int alignment) -> kernel_arg<It>
+{
+  kernel_arg<It> arg;
+  using T = value_t<It>;
+  cub::detail::uninitialized_copy_single(
+    &arg.template aliased_storage<aligned_base_ptr<const T>>(), make_aligned_base_ptr(ptr, alignment));
+  return arg;
+}
+
+// TODO(bgruber): make a variable template in C++14
+template <Algorithm Alg>
+using needs_aligned_ptr_t =
+  ::cuda::std::bool_constant<Alg == Algorithm::memcpy_async
+#ifdef _CUB_HAS_TRANSFORM_UBLKCP
+                             || Alg == Algorithm::ublkcp
+#endif // _CUB_HAS_TRANSFORM_UBLKCP
+                             >;
+
+template <Algorithm Alg, typename It, ::cuda::std::__enable_if_t<needs_aligned_ptr_t<Alg>::value, int> = 0>
+_CCCL_DEVICE auto select_kernel_arg(::cuda::std::integral_constant<Algorithm, Alg>,
+                                    kernel_arg<It>&& arg) -> aligned_base_ptr<const value_t<It>>&&
+{
+  return ::cuda::std::move(arg.template aliased_storage<aligned_base_ptr<const value_t<It>>>());
+}
+
+template <Algorithm Alg, typename It, ::cuda::std::__enable_if_t<!needs_aligned_ptr_t<Alg>::value, int> = 0>
+_CCCL_DEVICE auto select_kernel_arg(::cuda::std::integral_constant<Algorithm, Alg>, kernel_arg<It>&& arg) -> It&&
+{
+  return ::cuda::std::move(arg.template aliased_storage<It>());
+}
+
+// There is only one kernel for all algorithms, that dispatches based on the selected policy. It must be instantiated
+// with the same arguments for each algorithm. Only the device compiler will then select the implementation. This
+// saves some compile-time and binary size.
+template <typename MaxPolicy,
+          typename Offset,
+          typename F,
+          typename RandomAccessIteartorOut,
+          typename... RandomAccessIteartorsIn>
+__launch_bounds__(MaxPolicy::ActivePolicy::algo_policy::BLOCK_THREADS)
+  CUB_DETAIL_KERNEL_ATTRIBUTES void transform_kernel(
+    Offset num_items,
+    int num_elem_per_thread,
+    F f,
+    RandomAccessIteartorOut out,
+    kernel_arg<RandomAccessIteartorsIn>... ins)
+{
+  constexpr auto alg = ::cuda::std::integral_constant<Algorithm, MaxPolicy::ActivePolicy::algorithm>{};
+  transform_kernel_impl<typename MaxPolicy::ActivePolicy::algo_policy>(
+    alg,
+    num_items,
+    num_elem_per_thread,
+    ::cuda::std::move(f),
+    ::cuda::std::move(out),
+    select_kernel_arg(alg, ::cuda::std::move(ins))...);
+}
+
+constexpr int arch_to_min_bif(int sm_arch)
```

Comment:
suggestion: add a comment to explain that `bif` is bytes in flights

Reply:
Renamed to `arch_to_min_bytes_in_flight`.

Reply:
If you are referring to the work of @pciolkosz, then that's in `cudax` which is not useable in CUB. Where had you envisoned to put this function?

Reply:
yes, probably `cudax` is the best place for these constants

Reply:
Discussed in team VC yesterday: bytes in flight is rather a tuning parameter and should thus just stay here.